clang/lib/CodeGen/CGCall.cpp

```diff
@@ -732,6 +732,16 @@ CodeGenTypes::arrangeBuiltinFunctionDeclaration(CanQualType resultType,
                                  RequiredArgs::All);
 }
 
+const CGFunctionInfo &
+CodeGenTypes::arrangeSYCLKernelCallerDeclaration(QualType resultType,
+                                                 const FunctionArgList &args) {
+  auto argTypes = getArgTypesForDeclaration(Context, args);
```

Comment:
Will this place eventually create more changes to kernel signature in order to support sycl accessor type?

Reply:
No, the effects of parameter decomposition should already have been realized and be reflected in the `OutlinedFunctionDecl` declaration created for the `sycl_kernel_entry_point` attributed function. The commit message for commit https://github.com/llvm/llvm-project/commit/8fb42300a02c887740825cd1b60fc4fcd8d2f933 might be helpful to better explain the design intent.